src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1506,6 +1506,8 @@ def create_service_principal_for_rbac(
                                            ex.response.headers)  # pylint: disable=no-member
                     raise
 
+    logger.warning(CREDENTIAL_WARNING_MESSAGE)
```

Comment:
Print the warning earlier so that it is printed when `show_auth_for_sdk` is `True`.